src/reducers/SearchSlice.jsx

```diff
@@ -12,12 +12,20 @@ const initialState = {
   searchResults: [],
   searchValue: '',
   loading: 'idle',
+  currentRequestID: undefined,
+  error: null,
 };
 
 const search = createSlice({
   name: 'search',
   initialState,
   reducers: {
+    updateCurrentRequestID(state, action) {
+      state.currentRequestID = action.payload;
+    },
```

Comment:
My understanding is that this is used in tests only. At least you should state this in a comment, or possibly in the name of the action itself (eg `updateCurrentRequestIDTestOnly`). The best would be to avoid using test-only functions, but sometimes we can't avoid it :-)

Here is a general guiding principle we try to follow in the profiler: the tests should look like what the users would do.
Here are some more words about that: https://kentcdodds.com/blog/write-tests https://testing-library.com/docs/guiding-principles
This means less unit tests and more integration tests (that is: components with connected redux, and simulating user clicks).
This means stop testing things that can't happen in the real application.
This also means less test-specific code. It's not always possible to avoid that but this is what we're trying to do.
This approach has been quite successful for us, that's why I'm sharing it.

Hope this helps :)

Reply:
thanks! I appreciate all the feedback, this should be very helpful